src/EnergyPlus/IndoorIceRink.cc

```diff
@@ -0,0 +1,1031 @@
+                                               this->RefrigTempIn,
+                                               state.dataPlnt->PlantLoop(this->LoopNum).FluidIndex,
+                                               RoutineName);
+
+        int ConstrNum = DataSurfaces::Surface(SurfNum).Construction;
+        this->coeffs.Ca = RadSysTiHBConstCoef(SurfNum);
+        this->coeffs.Cb = RadSysTiHBToutCoef(SurfNum);
+        this->coeffs.Cc = RadSysTiHBQsrcCoef(SurfNum);
+
+        this->coeffs.Cd = RadSysToHBConstCoef(SurfNum);
+        this->coeffs.Ce = RadSysToHBTinCoef(SurfNum);
+        this->coeffs.Cf = RadSysToHBQsrcCoef(SurfNum);
+
+        this->coeffs.Cg = CTFTsrcConstPart(SurfNum);
+        this->coeffs.Ch = state.dataConstruction->Construct(ConstrNum).CTFTSourceQ(0);
+        this->coeffs.Ci = state.dataConstruction->Construct(ConstrNum).CTFTSourceIn(0);
+        this->coeffs.Cj = state.dataConstruction->Construct(ConstrNum).CTFTSourceOut(0);
+
+        this->coeffs.Ck =
+            this->coeffs.Cg + ((this->coeffs.Ci * (this->coeffs.Ca + this->coeffs.Cb * this->coeffs.Cd) +
+                                              this->coeffs.Cj * (this->coeffs.Cd + this->coeffs.Ce * this->coeffs.Ca)) /
+                                             (1.0 - this->coeffs.Ce * this->coeffs.Cb));
+        this->coeffs.Cl =
+            this->coeffs.Ch + ((this->coeffs.Ci * (this->coeffs.Cc + this->coeffs.Cb * this->coeffs.Cf) +
+                                              this->coeffs.Cj * (this->coeffs.Cf + this->coeffs.Ce * this->coeffs.Cc)) /
+                                             (1.0 - this->coeffs.Ce * this->coeffs.Cb));
+
+        this->RefrigTempIn = Node(RefrigNodeIn).Temp;
+        this->RefrigMassFlow = this->PastRefrigMassFlow;
+        this->Qsrcmax2 = this->PastRefrigMassFlow * CpRefrig * (this->RefrigTempIn - this->Tsrc);
+        this->Qsetpoint = ((((1 - (this->coeffs.Cb * this->coeffs.Ce)) * this->IceSetPointTemp) - this->coeffs.Ca - (this->coeffs.Cb * this->coeffs.Cd)) /
+             (this->coeffs.Cc + (this->coeffs.Cb * this->coeffs.Cf)));
+        this->ReqMassFlow = abs(this->Qsetpoint / ((this->Effectiveness * this->CpRefrig) * (this->RefrigTempIn - this->Tsrc)));
+
+        // Floor Surface temperatures. Only the current temperature is used. The others are for tracking and reporting purposes of the simulation.
+
+        
+
+        Real64 Volume = this->LengthRink * this->WidthRink * this->IceThickness;
+        this->RhoWater = FluidProperties::GetDensityGlycol(state, "WATER", this->IceTemperature, this->WaterIndex, RoutineName);
+        this->CpWater = FluidProperties::GetSpecificHeatGlycol(state, "WATER", this->IceTemperature, this->WaterIndex, RoutineName);
+        Real64 CpIce(2040.00);
+        Real64 RhoIce(917);
+        Real64 QFusion(333550.00);
+       
+
+
+        if (this->operation == 1) { // If schedule's value equals to 1 then Ice Rink is ON.
```

Comment:
What if the user schedules to something greater than 1?  And also maybe it's worth accounting for some variation in the floating point value.  I think a better pattern might be `if (this->operation > 0)`.